gulpfile.js

```diff
@@ -12,18 +20,95 @@ export const styles = () => {
     .pipe(plumber())
     .pipe(less())
     .pipe(postcss([
-      autoprefixer()
+      autoprefixer(),
+      csso()
     ]))
-    .pipe(gulp.dest('source/css', { sourcemaps: '.' }))
+    .pipe(rename('style.min.css'))
+    .pipe(gulp.dest('build/css', { sourcemaps: '.' }))
     .pipe(browser.stream());
 }
 
+//HTML
+
+const html = () => {
+  return gulp.src('source/*.html')
+    .pipe(htmlmin({ collapseWhitespace: true }))
+    .pipe(gulp.dest('build'));
+}
+
+//Scripts
+
+const scripts = () => {
+  return gulp.src('source/js/*.js')
+    .pipe(terser())
+    .pipe(gulp.dest('build/js'));
+}
+
+//Images
+
+const optimizeImages = () => {
+  return gulp.src('source/img/*.{jpg,png}')
+    .pipe(squoosh())
+    .pipe(gulp.dest('build/img'));
+}
+
+const copyImages = () => {
+  return gulp.src('source/img/*.{jpg,png}')
+    .pipe(gulp.dest('build/img'));
+}
+
+//WebP
+
+const createWebp = () => {
+  return gulp.src('source/img/*.{jpg,png}')
+    .pipe(squoosh({
+      webp: {}
+    }))
+    .pipe(gulp.dest('build/img'));
+}
+
+//SVG
+
+const svg = () => {
+  return gulp.src(['source/img/*.svg', '!source/img/icons/*.svg'])
```

Comment:
нам не нужно исключать поддиректории, так как первая строка масива означает только текущую директорию (без поддиректорий). Если бы нам нужно было учитывать текущую директорию с поддиректориями, мы бы писали вот так: `'source/img/**/*.svg'`
```suggestion
  return gulp.src(['source/img/*.svg'])
```